pkg/operator/ceph/file/filesystem.go

```diff
@@ -174,18 +176,18 @@ func newFS(name, namespace string) *Filesystem {
 }
 
 // SetPoolSize function sets the sizes for MetadataPool and dataPool
-func SetPoolSize(f *Filesystem, context *clusterd.Context, clusterInfo *client.ClusterInfo, spec cephv1.FilesystemSpec) error {
+func SetPoolSize(f *Filesystem, context *clusterd.Context, clusterInfo *client.ClusterInfo, spec cephv1.FilesystemSpec, clusterCrushRoot string) error {
 	// generating the metadata pool's name
 	metadataPoolName := generateMetaDataPoolName(f)
-	err := client.CreatePoolWithProfile(context, clusterInfo, metadataPoolName, spec.MetadataPool, "")
+	err := client.CreatePoolWithProfile(context, clusterInfo, metadataPoolName, spec.MetadataPool, "", clusterCrushRoot)
```

Comment:
Wherever we're passing the crush root into the pool creation methods, I have a request for a slightly different pattern. Instead of passing just the crush root, pass the `cephv1.ClusterSpec` type. This way, if we need other properties from the cluster spec in the future, the struct will already be easily available. We have done some other cleanup throughout Rook recently to pass the clusterSpec around, which has helped avoid having methods with too many parameters.

Reply:
@travisn As pointer/reference or directly?

Reply:
Ok, I went for "pass-by-value". I also had to move the GetCrushRootFromSpec to the client package in order to avoid a cyclic import.